taskotron_python_versions/common.py

```diff
@@ -69,6 +69,17 @@ def file_contains(path, needle):
             return mmf.find(needle) != -1
 
 
+def surrogate(text):
+    """Returns unicode for a given RPM header.
+    This is a layer to support both RPM 4.14 and 4.15.
+    RPM 4.14 always returns bytes.
+    RPM 4.15 always returns unicode strings with surrogateescaped errors.
+    """
+    if isinstance(text, bytes):
+        return text.decode('utf-8', errors='surrogate')
```

Comment:
I haven't found `surrogate` as a supported value, am I searching wrong?
https://docs.python.org/3/library/stdtypes.html#bytes.decode

Reply:
Good catch. it is surrogateescape, not surrogate and it is probably not properly documented:

```pycon
>>> 'ž'.encode('iso8859-2').decode('utf-8', errors='surrogateescape')
'\udcbe'
>>> 'ž'.encode('iso8859-2').decode('utf-8', errors='surrogate')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
LookupError: unknown error handler name 'surrogate'
```

Reply:
Actually, it is registered here https://docs.python.org/3/library/codecs.html#error-handlers

Reply:
I found it documented under:
https://docs.python.org/3/library/codecs.html#error-handlers

Reply:
Also reported this: https://bugs.python.org/issue37388